src/rpc/common/Validators.h

```diff
@@ -377,6 +377,36 @@ class IfType final
         validator_;
 };
 
+/**
+ * @brief A meta-validator that wrapp other validator to send the customized
+ * error
+ */
+template <typename Requirement>
+class WithCustomError final
+{
+    Requirement requirement;
+    RPC::Status error;
+
+public:
+    /**
+     * @brief Constructs a validator that calls the given validator "req" and
+     * return customized error "err"
+     */
+    WithCustomError(Requirement const& req, RPC::Status const& err)
```

Comment:
I think `Requirement&& req` is a better choice here in combination with `std::forward<Requirement>(req)`.
`err` could be passed by value or `rvalue ref` + `std::move` too. Value is probably better choice here for copy elision.

Reply:
Discuss offline, we will use value and move inside. Because we have some left value validators